modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -571,6 +571,140 @@ boolean pseudoClassStateChanged(PseudoClass pseudoClass) {
 
     }
 
+    // This method is a reduced version of transitionToState() method, it is added as a fix for JDK-8204568.
+    // Any modifications to the method transitionToStates() should be relatively applied here if needed.
+    // Currently this method is executed only by Labeled.fontProperty().set(), when it's font size
+    // is changed by LabeledText.
+    void recalculateRelativeSizeProperties(final Node node, Font fontForRelativeSizes) {
+
+        if (transitionStateInProgress) {
+            // If transitionToState() is being executed for the current control then all the css properties will get
+            // calculated there, and so we need to do anything here.
+            return;
+        }
+        if (cacheContainer == null) {
+            return;
+        }
+        final StyleMap styleMap = getStyleMap(node);
+        if (styleMap == null) {
+            return;
+        }
+        // if the style-map is empty, then we are only looking for inherited styles.
+        final boolean inheritOnly = styleMap.isEmpty();
+
+        final Set<PseudoClass>[] transitionStates = getTransitionStates(node);
+        CalculatedValue cachedFont = new CalculatedValue(fontForRelativeSizes, null, false);
+
+        final List<CssMetaData<? extends Styleable,  ?>> styleables = node.getCssMetaData();
+        final int numStyleables = styleables.size();
+
+        for (int n = 0; n < numStyleables; n++) {
+
+            final CssMetaData<Styleable,Object> cssMetaData =
+                    (CssMetaData<Styleable,Object>)styleables.get(n);
+
+            // Don't bother looking up styles that don't inherit.
+            if (inheritOnly && cssMetaData.isInherits() == false) {
+                continue;
+            }
+
+            // Skip the lookup if we know there isn't a chance for this property
+            // to be set (usually due to a "bind").
+            if (!cssMetaData.isSettable(node)) {
+                continue;
+            }
+
+            final String property = cssMetaData.getProperty();
+            boolean isFontProperty = property.equals("-fx-font") || property.equals("-fx-font-size");
+            // This method is executed as a result of change in font size of the control, hence font property should be skipped.
+            if (isFontProperty) {
+                continue;
+            }
+
+            CascadingStyle style = getStyle(node, property, styleMap, transitionStates[0]);
+            if (style != null) {
+                final ParsedValue cssValue = style.getParsedValue();
+                ObjectProperty<StyleOrigin> whence = new SimpleObjectProperty<>(style.getOrigin());
+                ParsedValue resolved = resolveLookups(node, cssValue, styleMap, transitionStates[0], whence, new HashSet<>());
+                boolean isRelative = ParsedValueImpl.containsFontRelativeSize(resolved, false);
+                if (!isRelative) {
```

Comment:
Is it possible to have a property that is absolute with sub-properties that might be relative? If so, then I think you need to add a check for `numSubProperties == 0` so you don't skip this case.

Reply:
I referred the [css ref](https://openjfx.io/javadoc/15/javafx.graphics/javafx/scene/doc-files/cssref.html) document which records the Available CSS Properties tables for different controls. I could find that only two css properties of region have sub properties `-fx-region-background` and `-fx-region-border`. No other control has a css property with sub properties. None of these two region css properties can be directly specified in css style but they are created using any of their sub properties if specified. Because these two properties cannot be set from css the code flow does not enter the `if (style != null)` block.
So it seems safe to me to proceed without sub properties check in the if block.
I also verified by printing the css properties and only two of these properties shown to have sub properties.